src/course-outline/configure-modal/VisibilityTab.jsx

```diff
@@ -4,18 +4,105 @@ import { Alert, Form } from '@edx/paragon';
 import { FormattedMessage, injectIntl } from '@edx/frontend-platform/i18n';
 import messages from './messages';
 
-const VisibilityTab = ({ isVisibleToStaffOnly, setIsVisibleToStaffOnly, showWarning }) => {
+const VisibilityTab = ({
+  isVisibleToStaffOnly,
+  setIsVisibleToStaffOnly,
+  showWarning,
+  isSubsection,
+  hideAfterDue,
+  setHideAfterDue,
+  showCorrectness,
+  setShowCorrectness,
+}) => {
   const handleChange = (e) => {
     setIsVisibleToStaffOnly(e.target.checked);
   };
 
+  const getVisibilityValue = () => {
+    if (isVisibleToStaffOnly) {
+      return 'hide';
+    }
+    if (hideAfterDue) {
+      return 'hideDue';
+    }
+    return 'show';
+  };
+
+  const visibilityChanged = (e) => {
+    const selected = e.target.value;
+    if (selected === 'hide') {
+      setIsVisibleToStaffOnly(true);
+      setHideAfterDue(false);
+    } else if (selected === 'hideDue') {
+      setIsVisibleToStaffOnly(false);
+      setHideAfterDue(true);
+    } else {
+      setIsVisibleToStaffOnly(false);
+      setHideAfterDue(false);
+    }
+  };
+
+  const correctnessChanged = (e) => {
+    setShowCorrectness(e.target.value);
+  };
+
   return (
     <>
-      <h3 className="mt-3"><FormattedMessage {...messages.sectionVisibility} /></h3>
+      {
+        isSubsection ? (
+          <h5 className="mt-4 text-gray-700"><FormattedMessage {...messages.subsectionVisibility} /></h5>
+        ) : (
+          <h5 className="mt-4 text-gray-700"><FormattedMessage {...messages.sectionVisibility} /></h5>
+        )
+      }
```

Comment:
You can use variables in messages instead of conditional rendering here. See [here](https://github.com/open-craft/frontend-app-course-authoring/blob/de0d8e5176e990bc85ade444098df654a51b1588/src/course-outline/delete-modal/DeleteModal.jsx#L18-L22) for an example.